homeassistant/components/config/entity_registry.py

```diff
@@ -184,11 +196,14 @@ def websocket_update_entity(hass, connection, msg):
         )
         return
 
-    result = {"entity_entry": _entry_ext_dict(entity_entry)}
+    result: dict[str, Any] = {"entity_entry": _entry_ext_dict(entity_entry)}
     if "disabled_by" in changes and changes["disabled_by"] is None:
         # Enabling an entity requires a config entry reload, or HA restart
-        config_entry = hass.config_entries.async_get_entry(entity_entry.config_entry_id)
-        if config_entry and not config_entry.supports_unload:
+        if (
+            not (entity_entry_id := entity_entry.config_entry_id)
+            or (config_entry := hass.config_entries.async_get_entry(entity_entry_id))
+            and not config_entry.supports_unload
+        ):
             result["require_restart"] = True
         else:
             result["reload_delay"] = config_entries.RELOAD_AFTER_UPDATE_DELAY
```

Comment:
Previous implementation returned `reload_delay` if the `entity_entry.config_entry_id` was `None`.
I think that was a bug, and I have adjusted the test accordingly.